Sources/PortalSwift/Components/PortalWebView.swift

```diff
@@ -413,6 +418,50 @@ public class PortalWebView: UIViewController, WKNavigationDelegate, WKScriptMess
     self.postMessage(payload: payload)
   }
 
+  private func handleWalletSwitchEthereumChain(method: String, params: [Any]) throws {
+    let encodedParams = try JSONSerialization.data(withJSONObject: params)
+    let params = try decoder.decode([ChainChangedParam].self, from: encodedParams)
+
+    print("⚠️ Params: \(params)")
+
+    let rawChainId = params[0].chainId
+
+    print("⚠️ Raw Chain ID: \(rawChainId)")
+    let chainId = try parseRawChainId(rawChainId: rawChainId)
+
+    print("⚠️ Chain ID: \(chainId)")
+
+    // Set the chainId locally
+    self.chainId = chainId
+
+    self.updateWebViewChain(chainId: chainId)
+
+    let jsonData = "[{ \"chainId\": \"\(rawChainId)\" }]"
+
+    let signatureReceivedJavascript = """
```

Comment:
why is handle switch ethereum chain sending the event `portal_signatureReceived` ?

Reply:
Because the Provider is no longer responsible for managing an active chain in v3.1+. So in this case, the WebView updates `self.chainId` and responds to the request.